irohad/ametsuchi/impl/flat_file/flat_file.cpp

```diff
@@ -16,132 +16,84 @@
  */
 
 #include "ametsuchi/impl/flat_file/flat_file.hpp"
-#include <dirent.h>
-#include <fstream>
+#include <boost/filesystem.hpp>
+#include <boost/filesystem/fstream.hpp>
+#include <iomanip>
+#include <sstream>
 #include "common/files.hpp"
 
 using namespace iroha::ametsuchi;
 
-const uint32_t DIGIT_CAPACITY = 16;
-
-// TODO 19/08/17 Muratov rework separator with platform independent approach
-// IR-495 #goodfirstissue
-const std::string SEPARATOR = "/";
-
-/**
- * Convert id to string repr
- * @param id - for conversion
- * @return string repr of identifier
- */
-std::string id_to_name(Identifier id) {
-  std::string new_id(DIGIT_CAPACITY, '\0');
-  std::sprintf(&new_id[0], "%016u", id);
-  return new_id;
-}
-
-/**
- * Convert string to identifier
- * @param name - string for conversion
- * @return numeric identifier
- */
-Identifier name_to_id(const std::string &name) {
-  std::string::size_type sz;
-  return static_cast<Identifier>(std::stoul(name, &sz));
-}
-
-/**
- * Check if file exists
- * @param name - full path to file
- * @return true, if exists
- */
-bool file_exist(const std::string &name) {
-  struct stat buffer {};
-  return stat(name.c_str(), &buffer) == 0;
-}
-
-/**
- * Remove file from folder
- * @param dump_dir - target dir
- * @param id - identifier of file
- */
-void remove(const std::string &dump_dir, std::string filename) {
-  auto f_name = dump_dir + SEPARATOR + filename;
-
-  if (std::remove(f_name.c_str()) != 0) {
-    logger::log("FLAT_FILE")
-        ->error("remove({}, {}): error on deleting file", dump_dir, filename);
+namespace {
```

Comment:
why nameless namespace?

Reply:
Anonymous namespace makes the symbol translation-unit local and prevents someone outside the translation unit accidentally calling the function.

Since these functions are implementation details I put them into an anonymous namespace.

http://www.goldsborough.me/c/c++/linker/2016/03/30/19-34-25-internal_and_external_linkage_in_c++/

Reply:
Ah, just noticed it doesn't cover all file, sorry, it's fine then